### PR TITLE
[Doc] Add *processing guarantee for connector*

### PR DESCRIPTION
**Pull request by Anonymitaet**
Fix https://github.com/apache/pulsar/issues/5015

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@tuteng could you please help review? Thank you

File: site2/docs/io-overview.md [hunk 1]
```diff
@@ -34,9 +34,106 @@ Common sinks include other messaging systems and SQL and NoSQL databases.
 
 For the complete list of Pulsar built-in sink connectors, see [sink connector](io-connectors.md#sink-connector).
 
+## Processing guarantee
+
+Processing guarantees are used to:
+
+* handle errors when writing messages to Pulsar topics.
+  
+* determine the subscription type used by Pulsar when reading messages from Pulsar topics.
```
Comment (tuteng) [inline site2/docs/io-overview.md]: I don't think this part of the content is necessary, and there is no necessary relationship between `Processing guarantees` and `subscription type`.

Comment (Anonymitaet) [reply site2/docs/io-overview.md]: @tuteng thank you. I've incorporated your comment, could you PTAL?

**Comment by Anonymitaet** [comment]
run java8 tests

**Pull request merged**